src/features/deliveryStatus/useMessageDeliveryStatus.tsx

```diff
@@ -1,57 +1,83 @@
 import { useQuery } from '@tanstack/react-query';
-import { useEffect, useMemo } from 'react';
+import { useEffect } from 'react';
 import { toast } from 'react-toastify';
 
+import { MultiProvider } from '@hyperlane-xyz/sdk';
 import { errorToString } from '@hyperlane-xyz/utils';
 
-import { useMultiProvider, useRegistry } from '../../store';
+import { useReadyMultiProvider, useRegistry } from '../../store';
 import { Message, MessageStatus } from '../../types';
 import { logger } from '../../utils/logger';
 import { MissingChainConfigToast } from '../chains/MissingChainConfigToast';
 import { useChainConfigs } from '../chains/useChainConfigs';
+import { isEvmChain } from '../chains/utils';
 
 import { fetchDeliveryStatus } from './fetchDeliveryStatus';
 
-export function useMessageDeliveryStatus({ message, pause }: { message: Message; pause: boolean }) {
+export function useMessageDeliveryStatus({
+  message,
+  enabled = true,
+}: {
+  message: Message;
+  enabled: boolean;
+}) {
   const chainConfigs = useChainConfigs();
-  const multiProvider = useMultiProvider();
+  const multiProvider = useReadyMultiProvider();
   const registry = useRegistry();
 
-  const serializedMessage = JSON.stringify(message);
-  const { data, error, isFetching } = useQuery(
-    ['messageDeliveryStatus', serializedMessage, pause],
-    async () => {
-      if (pause || !message || message.status === MessageStatus.Delivered) return null;
+  const { data, error, isFetching } = useQuery({
+    queryKey: ['messageDeliveryStatus', message, !!multiProvider],
+    queryFn: async () => {
+      if (!multiProvider || message.status == MessageStatus.Delivered) {
+        return { message };
+      }
```

Comment:
can we/should we fall back to JSON RPC queries?

Reply:
seems like a quite lightweight query here

Reply:
SmartProvider retries and fallsback already and this whole query runs on a 10s interval